### PR TITLE
Add NEWS and README

### PR DESCRIPTION
**Pull request by anirudh2290**
## Description ##
Adding NEWS and README

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by anirudh2290** [comment]
I have addressed the comments in https://github.com/apache/incubator-mxnet/pull/11267

File: python/mxnet/gluon/block.py [hunk 1]
```diff
@@ -304,24 +305,58 @@ def _collect_params_with_prefix(self, prefix=''):
             ret.update(child._collect_params_with_prefix(prefix + name))
         return ret
 
-    def save_params(self, filename):
+    def save_parameters(self, filename):
         """Save parameters to file.
```
Comment (piiswrong) [inline python/mxnet/gluon/block.py]:
```
Save parameters to file.

Saved parameters can only be loaded with `load_parameters`. Note that this
method only saves parameters, not model structure. If you want to save model
structures, please use :py:meth:`HybridBlock.export`.

Parameters
-----------
...

References
----------
  `Saving and Loading Gluon Models
  <https://mxnet.incubator.apache.org/tutorials/gluon/save_load_params.html>`_
```

File: python/mxnet/gluon/block.py [hunk 2]
```diff
@@ -304,24 +305,58 @@ def _collect_params_with_prefix(self, prefix=''):
...
+        warnings.warn("save_params is deprecated. Please use save_parameters.")
+        try:
+            self.collect_params().save(filename, strip_prefix=self.prefix)
+        except ValueError as e:
+            raise ValueError('%s\nsave_params is deprecated. Using ' \
+                              'save_parameters may resolve this error.'%e.message)
+
+    def load_parameters(self, filename, ctx=None, allow_missing=False,
+                        ignore_extra=False):
         """Load parameters from file.
```
Comment (piiswrong) [inline python/mxnet/gluon/block.py]:
```
Load parameter file previously saved by `save_parameters`.

Parameters
-----------
...

References
----------
  `Saving and Loading Gluon Models
  <https://mxnet.incubator.apache.org/tutorials/gluon/save_load_params.html>`_
```

**Comment by piiswrong** [comment]
This is API manual, it should be succinct. Mentioning too many things will only confuse users.

**Comment by anirudh2290** [comment]
@piiswrong I have made the changes as suggested.

File: docs/tutorials/gluon/save_load_params.md [hunk 3]
```diff
@@ -0,0 +1,261 @@
+# Saving and Loading Gluon Models
```
Comment (piiswrong) [inline docs/tutorials/gluon/save_load_params.md]: Please change save_params/load_params to save_parameters/load_parameters

**Pull request merged**